chain/network/src/network_protocol/tests.rs

```diff
@@ -170,3 +176,96 @@ fn serialize_deserialize() -> anyhow::Result<()> {
+    // Convert proto::IpAddr back to IpAddr to be independent of protocol buffer in near network
+    let result_deserialized_actual_ip_addr = net::IpAddr::try_from(&proto_ip_addr_after_deserialize);
+    assert!(result_deserialized_actual_ip_addr.is_ok());
+    let deserialized_actual_ip_addr = result_deserialized_actual_ip_addr.unwrap();
+    assert_eq!(ip_addr_before_serialize, deserialized_actual_ip_addr);
+}
+
+#[test]
+fn sign_and_verify_ip_addr_with_peer_id() {
+    // Design sign and verify algorithm for std::net::IpAddr
+    let seed = "123";
+    let node_key = SecretKey::from_seed(KeyType::ED25519, seed);
+    let peer_id = PeerId::new(node_key.public_key());
+    let mut rng = make_rng(89028037453);
+    let ip_addr: net::IpAddr = data::make_ipv4(&mut rng);
+    let ip_bytes: Vec<u8> = match ip_addr {
+        net::IpAddr::V4(ip) => ip.octets().to_vec(),
+        net::IpAddr::V6(ip) => ip.octets().to_vec(),
+    };
+    let signature = node_key.sign(&ip_bytes);
+    assert!(signature.verify(&ip_bytes, &node_key.public_key()));
+    assert!(signature.verify(&ip_bytes, peer_id.public_key()));
+
+    // Wrap ip address sign and verify algorithm with interfaces: OwnedIpAddress and SignedOwnedIpAddress
+    let owned_ip_address = OwnedIpAddress {
+        ip_address: ip_addr.clone(),
+    };
+    let signed_owned_ip_address: SignedOwnedIpAddress = owned_ip_address.sign(&node_key);
+    assert!(signed_owned_ip_address.verify(&node_key.public_key()));
+    assert!(signed_owned_ip_address.verify(peer_id.public_key()));
+}
+
+#[test]
+fn serialize_deserialize_owned_ip_address() {
+    let mut rng = make_rng(89028037453);
+    let ip_addr: net::IpAddr = data::make_ipv4(&mut rng);
+    let owned_ip_addr_before_serialize = OwnedIpAddress {
+        ip_address: ip_addr.clone(),
+    };
+
+    // Convert OwnedIpAddress to proto::OwnedIpAddr for protocol buffer serialization
+    let proto_owned_ip_addr_before_serialize: proto::OwnedIpAddr = proto::OwnedIpAddr::from(&owned_ip_addr_before_serialize);
+    // Serialize and deserialize for transporting over the network
+    let owned_ip_addr_bytes: Vec<u8> = proto_owned_ip_addr_before_serialize.write_to_bytes().unwrap();
+    let proto_owned_ip_addr_after_deserialize: proto::OwnedIpAddr = proto::OwnedIpAddr::parse_from_bytes(&owned_ip_addr_bytes).unwrap();
+    assert_eq!(proto_owned_ip_addr_before_serialize,  proto_owned_ip_addr_after_deserialize);
+    // Convert proto::OwnedIpAddr back to OwnedIpAddress to be independent of protocol buffer in near network
+    let result_deserialized_actual_owned_ip_addr = OwnedIpAddress::try_from(&proto_owned_ip_addr_after_deserialize);
+    assert!(result_deserialized_actual_owned_ip_addr.is_ok());
+    let deserialized_actual_owned_ip_addr = result_deserialized_actual_owned_ip_addr.unwrap();
+    assert_eq!(owned_ip_addr_before_serialize, deserialized_actual_owned_ip_addr);
+}
+
+#[test]
+fn serialize_deserialize_signed_owned_ip_address() {
```

Comment:
This is a reasonable thing to test, but it is outside the scope of this file. Maybe you could add another test in `connection_pool.rs` testing the happy case (you have two failure cases)?

Reply:
I spent 4 hours trying to test the happy case, but it just doesnt terminate. I'm not sure if this code allows testing the happy case. I've tested the happy case in other files (peer/communication.rs and integration-tests/network/peer_handshake.rs

Reply:
maintain this test since it doesnt go through the network and is mainly specific to protobuf serialization, in which is tested in this specific file.

Reply:
Resolving conversation as it seems this test was removed.